tapeout/src/main/scala/transforms/AddSuffixToModuleNames.scala

```diff
@@ -18,30 +19,36 @@ case class ModuleNameSuffixAnnotation(target: CircuitTarget, suffix: String)
   def duplicate(n: CircuitTarget) = this.copy(target = n)
 }
 
-// This doesn't rename ExtModules under the assumption that they're some
-// Verilog black box and therefore can't be renamed.  Since the point is to
-// allow FIRRTL to be linked together using "cat" and ExtModules don't get
-// emitted, this should be safe.
-class AddSuffixToModuleNames extends Transform {
-  def inputForm = LowForm
-  def outputForm = LowForm
+class AddSuffixToModuleNames extends Transform with DependencyAPIMigration {
 
-  def processAnnos(annos: AnnotationSeq): (AnnotationSeq, (String) => String) = {
-    val whitelist = annos.collect({ case KeepNameAnnotation(tgt) => tgt.module }).toSet
-    val newAnnos = annos.filterNot(_.isInstanceOf[ModuleNameSuffixAnnotation])
-    val suffixes = annos.collect({ case ModuleNameSuffixAnnotation(_, suffix) => suffix })
-    require(suffixes.length <= 1)
+  override def prerequisites: Seq[TransformDependency] = Forms.LowForm
+  override def optionalPrerequisites: Seq[TransformDependency] = Forms.LowFormOptimized
+  override def optionalPrerequisiteOf: Seq[TransformDependency] = Forms.LowEmitters
+  override def invalidates(a: Transform): Boolean = false
 
+  def determineRenamerandAnnos(state: CircuitState): (AnnotationSeq, (String) => String) = {
+    // remove determine suffix annotation
+    val newAnnos = state.annotations.filterNot(_.isInstanceOf[ModuleNameSuffixAnnotation])
+    val suffixes = state.annotations.collect({ case ModuleNameSuffixAnnotation(_, suffix) => suffix })
+    require(suffixes.length <= 1)
     val suffix = suffixes.headOption.getOrElse("")
-    val renamer = { name: String => if (whitelist(name)) name else name + suffix }
+
+    // skip renaming ExtModules and top-level module
```

Comment:
One notable change is that this transform automatically skips renaming `ExtModule`s and the `state.circuit.main` module. This allows you to not have a list of `externals` that shouldn't be renamed in the `Generate.scala` file.